### PR TITLE
Feature/crud

### PR DESCRIPTION
**Pull request by Beardless-sheik**
In this pull request: 
- Remove all hard coded tasks
- create adding, editing, clear all completed and removal of tasks in another module and imported into main index.js for functionality 
- default new tasks set to false and indexes updated after each user action
- changes to to do list are saved in local storage for persistence

### CODE REVIEW

File: src/crud.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+      const index = tasksArray.findIndex((element) => element.index === parseInt(inputWithId.id, 10));
+      tasksArray[index].description = target.innerHTML;
+      localStorage.setItem('tasksList', JSON.stringify(tasksArray));
+    }
+  });
+}
+
+const representNewIndexes = (tasksArray) => {
+  // eslint-disable-next-line no-plusplus
+  for (let i = 0; i < tasksArray.length; i++) {
```
Comment (Emmyn5600) [inline src/crud.js]: - [ ] Let us fix eslint errors without disabling them, this makes your code looks unprofessional.

File: src/index.js [hunk 2]
```diff
@@ -1,63 +1,123 @@
...
+    const { target } = event;
+    if (target.classList.value.indexOf('delete') !== -1) {
+      const tasksArray = JSON.parse(localStorage.getItem('tasksList'));
+      // eslint-disable-next-line max-len
+      const index = tasksArray.findIndex((element) => element.index === parseInt(target.id, 10));
+      // deleteTask(index, tasksArray);
+      // representNewIndexes(tasksArray);
+      tasksArray.splice(index, 1);
+      localStorage.setItem('tasksList', JSON.stringify(tasksArray));
+      // location.reload();
```
Comment (Emmyn5600) [inline src/index.js]: - [ ]  Let us fix eslint errors without disabling them, this makes your code looks unprofessional.
- [ ]  Remove the commented lines in the code that has been pushed.

File: src/index.js [hunk 3]
```diff
@@ -1,63 +1,123 @@
...
+    location.reload();
+  }
+});
+
+bookList.addEventListener('click', (event) => {
+  const { target } = event;
+  if (target.classList.value.indexOf('delete') !== -1) {
+    const tasksArray = JSON.parse(localStorage.getItem('tasksList'));
+    // eslint-disable-next-line max-len
+    const index = tasksArray.findIndex((element) => element.index === parseInt(target.id, 10));
```
Comment (Emmyn5600) [inline src/index.js]: - [ ] Let us fix eslint errors without disabling them, this makes your code looks unprofessional.

File: src/status.js [hunk 4]
```diff
@@ -1,8 +1,12 @@
...
     tasksArray[index].completed = !tasksArray[index].completed;
-    localStorage.setItem('TasksList', JSON.stringify(tasksArray));
+    localStorage.setItem('tasksList', JSON.stringify(tasksArray));
   });
-}
\ No newline at end of file
+}
+
+// eslint-disable-next-line import/prefer-default-export
+export { checkBoxEventListener };
```
Comment (Emmyn5600) [inline src/status.js:12]: - [ ] Let us fix eslint errors without disabling them, this makes your code looks unprofessional.

File: src/crud.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+      const tasksArray = JSON.parse(localStorage.getItem('tasksList'));
+      const filteredArray = tasksArray.filter((element) => element.completed === false);
+      representNewIndexes(filteredArray);
+      localStorage.setItem('tasksList', JSON.stringify(filteredArray));
+      // eslint-disable-next-line no-restricted-globals
+      location.reload();
+    }
+  });
+};
+
```
Comment (Emmyn5600) [inline src/crud.js]: - [ ] Adding, deleting and editing a task function doesn't work, please work on it

rr20

- [ ] Your webpack.config.js file seem to have an issue, kindly check the configuration again to fix that issue.

rr19

**Review by Emmyn5600 — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUESTED :recycle:

Hi, @Beardless-sheik 

Great work on implementing the changes requested by the previous reviewer, however there are some issues to be solved before you get approved but you are almost there:muscle:
   
 - The repo looks professional.:heavy_check_mark:
- There are no linter errors:heavy_check_mark:
- You have descriptive PR:heavy_check_mark:

Changes required :writing_hand: 

         - Please work on the review too

    Cheers and Happy coding!:clap::clap::clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### _**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_
------------------------------------------------------------------------------------------------------------------------------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

**Comment by Beardless-sheik** [comment]
Hey @Emmyn5600  Thanks for the input. Kindly explain what doesn't work with the "adding, deletion, functionality". It seems to work on my local dev. What part of my webpack doesn't work? It seems to just give me "warnings" about bundle size?

**Review by mikenath223 — APPROVED** [review]
Hello there  

Your project is complete!  There is nothing else to say other than... it's time to merge it 🛳 
Congratulations! 🎉 

Happy coding. Cheers! 🚀🚀

If you feel something isn't clear enough you can leave a question or comment in the PR thread.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**